components/storage/blockdevices/COMPONENT_SD/config/mbed_lib.json

```diff
@@ -0,0 +1,70 @@
+{
+    "name": "sd",
+    "config": {
+        "SPI_CS": "SPI_PERSISTENT_MEM_CS",
+        "SPI_MOSI": "SPI_MOSI",
+        "SPI_MISO": "SPI_MISO",
+        "SPI_CLK": "SPI_SCK",
+        "DEVICE_SPI": 1,
+        "FSFAT_SDCARD_INSTALLED": 1,
+        "CMD_TIMEOUT": 10000,
+        "CMD0_IDLE_STATE_RETRIES": 5,
+        "SD_INIT_FREQUENCY": 100000
+    },
+    "target_overrides": {
+        "DISCO_L476VG": {
+          "SPI_MOSI": "PE_15",
+          "SPI_MISO": "PE_14",
+          "SPI_CLK":  "PE_13",
+          "SPI_CS":   "PE_12"
+        },
+        "LPC2368": {
+             "SPI_MOSI": "p11",
+             "SPI_MISO": "p12",
+             "SPI_CLK":  "p13",
+             "SPI_CS":   "p14"
+        },
+        "NUCLEO_F411RE": {
+             "SPI_MOSI": "PC_3",
```

Comment:
Why didn't you keep default SPI_MOSI/MISO pins ?

Reply:
Actually, because this configuration was imported from the sd-driver repo. if it stays here then it implies that there might be a conflict between PinName and the config file. 
I really don't know right now which one is the correct one, so I left it as it was for backward compatibility.